src/librustc/hir/mod.rs

```diff
@@ -1356,7 +1356,6 @@ impl Expr {
             ExprKind::Unary(..) => ExprPrecedence::Unary,
             ExprKind::Lit(_) => ExprPrecedence::Lit,
             ExprKind::Type(..) | ExprKind::Cast(..) => ExprPrecedence::Cast,
-            ExprKind::If(..) => ExprPrecedence::If,
```

Comment:
Can this `Precedence` variant be removed now or is it needed for the AST?

Reply:
I believe AST has its own https://doc.rust-lang.org/nightly/nightly-rustc/syntax/ast/struct.Expr.html#method.precedence so it shouldn't be needed.